### PR TITLE
Reversion of reversion of Max's improvements

### PR DESCRIPTION
**Pull request by robinbb**

### CODE REVIEW

**Comment by jstavh** [comment]
It's generally considered against Rust naming conventions to use fully capitalized names or to have capitalized acronyms within CamelCase names. According to Rust's naming conventions (RFC 430), acronyms should be treated as a single word in CamelCase. For example, instead of `HTTPResponse`, the convention suggests using `HttpResponse`.
- For further details, see https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/upper_case_acronyms.rs
- For how Mina sees ERC, see https://github.com/MinaProtocol/MIPs

File: server/src/models/diesel.rs [hunk 1]
```diff
@@ -17,7 +17,7 @@ pub(crate) enum ProposalCategory {
     Core,
     Networking,
     Interface,
-    ERC,
+    Erc,
```
Comment (robinbb) [inline server/src/models/diesel.rs:20]: I suspect that this Enum (ProposalCategory) is going to be written verbatim to the UI. In that case, "ERC" should indeed match the documentation elsewhere (Mina's MIP repo). In that case, perhaps the pragma that tells the linter to allow ERC (and not require "Erc") should indeed be used, here, preferably locally and not merely on this entire file. Alternately, you could explicity specify how to "show" or "serialize" these things, such that "Erc" serializes to "ERC". In any case, consistency with the MIP documentation is important, and takes precedence over the Rust naming conventions that the linter is trying to ensure.

**Comment by 52** [comment]
@jstavh FYI - to fix the failing web CI you need to force dynamic rendering for the root page by adding the following export to `web/app/page.tsx`:

```ts
export const dynamic = 'force-dynamic';
```

So it should be:
```ts
...
import { ProposalTable } from 'components/proposal-table';

export const dynamic = 'force-dynamic';

const Page = async () => {
  const proposals = await getProposalList();
...
```

You then need to revert the commits https://github.com/Granola-Team/mina-on-chain-voting/pull/242/commits/d8ee985c94922d5e134b2302bb63bbd58d420aa7 & https://github.com/Granola-Team/mina-on-chain-voting/pull/242/commits/5e0ce11d25284345b8f4658985f83600199bc85f.

File: server/src/models/diesel.rs (hunk 1 above)
Comment (jstavh) [reply server/src/models/diesel.rs]: @robinbb I reverted and and the macro https://github.com/Granola-Team/mina-on-chain-voting/pull/242/commits/97cbc0c548000d343365f568214055da19b5957e and https://github.com/Granola-Team/mina-on-chain-voting/pull/242/commits/baa8705b9eae632ba0564b89c56e2a1b7791c4dc

**Review by jstavh — APPROVED** [review]
Approving these changes based on conversations in the PR.

**Comment by jstavh** [comment]
@52 your method works better so I did as you wrote and the web CI passes.

**Comment by jstavh** [comment]
> @jstavh FYI - to fix the failing web CI you need to force dynamic rendering for the root page by adding the following export to `web/app/page.tsx`:
> 
> ```ts
> export const dynamic = 'force-dynamic';
> ```
> 
> So it should be:
> 
> ```ts
> ...
> import { ProposalTable } from 'components/proposal-table';
> 
> export const dynamic = 'force-dynamic';
> 
> const Page = async () => {
>   const proposals = await getProposalList();
> ...
> ```
> 
> You then need to revert the commits [d8ee985](https://github.com/Granola-Team/mina-on-chain-voting/commit/d8ee985c94922d5e134b2302bb63bbd58d420aa7) & [5e0ce11](https://github.com/Granola-Team/mina-on-chain-voting/commit/5e0ce11d25284345b8f4658985f83600199bc85f).


@52 your method works better so I did as you wrote and the web CI passes.

**Comment by jstavh** [comment]
@52 your method works better so I did as you wrote and the web CI passes.

**Comment by robinbb** [comment]
@jstavh I recommend rewriting the Git history so that the reversions are explicit but the other changes are collapsed into 1 or 2 commits (with 'git rebase -i' for example), before merging in order to have a better commit history.

**Pull request merged**